### PR TITLE
support for RN >= 0.56

### PR DESCRIPTION
**Pull request by jesucarr**

### CODE REVIEW

**Comment by jesucarr** [comment]
Would you have a look into this and merge?

**Comment by jesucarr** [comment]
Yes `babelTransformerPath: require.resolve("./transformer")` is in line 48 https://github.com/javascript-obfuscator/react-native-obfuscating-transformer/pull/7/files#diff-f48aafff6e296a5dab713b5ffbce17f5R48

I just commited the updated documentation

File: README.md [hunk 1]
```diff
@@ -16,8 +16,8 @@ or
 
 ```diff
  module.exports = {
-+  getTransformModulePath() {
-+    return require.resolve("./transformer")
++  transformer {
```
Comment (marsinearth) [inline README.md]: transformer: {

missed a colon

**Comment by marsinearth** [comment]
I used this pull request with yarn add javascript-obfuscator/react-native-obfuscating-transformer#7/head but keep getting `cannot fine module 'react-native-obfuscating-transformer'` when react-native run-ios

**Comment by krszwsk** [comment]
@marsinearth it doesn't work since you have to compile typescript first. Checkout to this branch, run `yarn`, `tsc`, and you should be good to go.

**Comment by sytler** [comment]
@krszwsk so the upstream transformer is broken here?

**Comment by krszwsk** [comment]
No, you just have to compile TS first in order to use the package from the repository directly.

**Comment by sytler** [comment]
@krszwsk have you tried this transformer on 0.57.7+?

**Comment by krszwsk** [comment]
@sytler yup, I ran it on 0.57.8

**Comment by sytler** [comment]
Im using this config and somehow it is not working

**rn-cli.config.js**
```
const { getDefaultConfig } = require("metro-config");

module.exports = (async () => {
  const {
    resolver: { sourceExts }
  } = await getDefaultConfig();
  return {
    transformer: {
      babelTransformerPath: require.resolve('./transformer'),
    },
    resolver: {
      sourceExts: [...sourceExts]
    }
  };
})();
```

**transformer.js**

```
const obfuscatingTransformer = require('react-native-obfuscating-transformer');
const typescriptTransformer = require('react-native-typescript-transformer');

module.exports = obfuscatingTransformer({
  upstreamTransformer: typescriptTransformer,
  obfuscatorOptions: {
    compact: true,
    // controlFlowFlattening: true,
    // controlFlowFlatteningThreshold: 0.75,
    // deadCodeInjection: true,
    // deadCodeInjectionThreshold: 0.4,
    debugProtection: false,
    disableConsoleOutput: true,
    identifierNamesGenerator: "hexadecimal",
    log: true,
    reservedStrings: ['import', 'require', 'class declaration', 'export'],
    rotateStringArray: true,
    seed: 1,
    selfDefending: false, // only false - in another case - break application!
    sourceMap: true,
    sourceMapMode: "separate",
    stringArray: true,
    stringArrayEncoding: 'base64',
    stringArrayThreshold: 0.8,
    transformObjectKeys: false, // only false - in another case - break build process!
    unicodeEscapeSequence: true,
  },
  trace: true,
  emitObfuscatedFiles: false,
  enableInDevelopment: true,
});
```

**error**

SyntaxError in src/api/client/index.ts: Unexpected token, expected , (1:3748)

SyntaxError: Unexpected token, expected , (1:3748)
    at Parser.pp$5.raise (/Users/user/Desktop/Projects/REPO/project/node_modules/babylon/lib/index.js:4454:13)
    at Parser.pp.unexpected (/Users/user/Desktop/Projects/REPO/project/node_modules/babylon/lib/index.js:1761:8)
    at Parser.pp.expect (/Users/user/Desktop/Projects/REPO/project/node_modules/babylon/lib/index.js:1749:33)
    at Parser.pp$3.parseExprList (/Users/user/Desktop/Projects/REPO/project/node_modules/babylon/lib/index.js:4296:12)
    at Parser.pp$3.parseNew (/Users/user/Desktop/Projects/REPO/project/node_modules/babylon/lib/index.js:3916:27)
    at Parser.pp$3.parseExprAtom (/Users/user/Desktop/Projects/REPO/project/node_modules/babylon/lib/index.js:3733:19)
    at Parser.pp$3.parseExprSubscripts (/Users/user/Desktop/Projects/REPO/project/node_modules/babylon/lib/index.js:3494:19)
    at Parser.pp$3.parseMaybeUnary (/Users/user/Desktop/Projects/REPO/project/node_modules/babylon/lib/index.js:3474:19)
    at Parser.pp$3.parseExprOps (/Users/user/Desktop/Projects/REPO/project/node_modules/babylon/lib/index.js:3404:19)
    at Parser.pp$3.parseMaybeConditional (/Users/user/Desktop/Projects/REPO/project/node_modules/babylon/lib/index.js:3381:19)

**Pull request merged**